pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -408,14 +418,58 @@ def create(self, request, path):
 
         data = {
             "task": reverse(
-                "collection-imports-detail",
-                kwargs={"path": path, "pk": async_result.pk},
+                settings.ANSIBLE_URL_NAMESPACE + "collection-imports-detail",
+                kwargs={"pk": async_result.pk},
                 request=None,
             )
         }
         return Response(data, status=http_status.HTTP_202_ACCEPTED)
 
 
+class LegacyCollectionUploadViewSet(CollectionUploadViewSet):
+    """
+    Collection upload viewset with deprecated markers for the openAPI spec.
+    """
+
+    @extend_schema(
+        description="Create an artifact and trigger an asynchronous task to create "
+        "Collection content from it.",
+        summary="Upload a collection",
+        request=CollectionOneShotSerializer,
+        responses={202: AsyncOperationResponseSerializer},
+        deprecated=True,
+    )
+    def create(self, request, path):
+        return super().create(request, distro_base_path=path)
+
+
+class CollectionArtifactDownloadView(views.APIView):
+    action = "download"
+
+    def urlpattern(*args, **kwargs):
+        return "ansible-collection-download-viewset"
+
+    def get(self, request, *args, **kwargs):
+        distro_base_path = self.kwargs["distro_base_path"]
+        filename = self.kwargs["filename"]
+        prefix = settings.CONTENT_PATH_PREFIX.strip("/")
+        distribution = AnsibleDistribution.objects.get(base_path=distro_base_path)
+
+        url = "{host}/{prefix}/{distro_base_path}/{filename}".format(
+            host=settings.CONTENT_ORIGIN.strip("/"),
+            prefix=prefix,
```

Comment:
could both of these params pull from settings here to make it easier to read?
```suggestion
        distribution = AnsibleDistribution.objects.get(base_path=distro_base_path)

        url = "{host}/{prefix}/{distro_base_path}/{filename}".format(
            host=settings.CONTENT_ORIGIN.strip("/"),
            prefix=settings.CONTENT_PATH_PREFIX.strip("/"),
```

Reply:
done